app/src/main/java/com/tranhoabinh/framgia/moviedbkotlin/core/BaseFragment.kt

```diff
@@ -37,14 +38,28 @@ abstract class BaseFragment<ViewBinding : ViewDataBinding, ViewModel : BaseViewM
         }
     }
 
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        viewModel.apply {
+            errorMessage.observe(this@BaseFragment, Observer {
+                if (!it.isNullOrEmpty()) {
+                    showToast(it)
+                    errorMessage.value = ""
```

Comment:
Chỗ này dùng livedata a làm thế này em nghĩ là khi xuất hiện lỗi nó sẽ show lỗi lên và sau đó show một cái Toast không có gì lên vì a đã set lại giá trị cho erroMessage. A chạy thử lại xem đúng không?

Reply:
if (!it.isNullOrEmpty()) {
@hoangth1 ko show cái rỗng đâu em có đoạn check rồi, nhưng code như này ko hay lắm

Reply:
cái này liên quan đến singleliveevent https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150. Nhưng có nên thêm 1 class singleliveevent cho đoạn code này k. Mình thấy cách 2 cũng đáp ứng đc như cầu nên dùng cách 2 luôn

Reply:
à, em không để ý cái isNullOrEmpty()